subprojects/analysis-kotlin-api/src/main/kotlin/org/jetbrains/dokka/analysis/kotlin/documentable/ExternalDocumentableProvider.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package org.jetbrains.dokka.analysis.kotlin.documentable
+
+import org.jetbrains.dokka.DokkaConfiguration.DokkaSourceSet
+import org.jetbrains.dokka.links.DRI
+import org.jetbrains.dokka.model.DClasslike
+
+/**
+ * TODO [external-documentable-provider] add documentation
+ */
+public interface ExternalDocumentableProvider {
```

Comment:
In a few words, this extension is used in dackka to create documentables for supertypes that come from dependencies, because, if I'm not mistaken, by default Dokka only creates the documentable model for the code it finds in the project itself, but not its classpath.

Reply:
BTW, what about typealiases? They are not `DClasslike` but could be supertypes, and so do we need to support them?

Reply:
It could indeed be useful, but we'd have to think about the corner cases, write tests for it and make sure the implementations support it, and at this point it doesn't seem to be needed, at least not for #3112. If someone does come with such a request, we can certainly consider it

Reply:
> BTW, what about typealiases? They are not DClasslike but could be supertypes, and so do we need to support them?

AFAIK, [dokka-devsite-plugin](https://android.googlesource.com/platform/tools/dokka-devsite-plugin) can use `TypeAliased` that has `inner`.